### PR TITLE
add BubbleStorageIterator

### PR DESCRIPTION
**Pull request by calvinnhieu**

### CODE REVIEW

File: bust-a-move-framework/src/main/java/bust_a_move/util/Iterator.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
+ * File: Interface.java
```
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/bust_a_move/util/Iterator.java]: file: iterator.java

File: bust-a-move-framework/src/main/java/bust_a_move/util/Iterator.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * @author Calvin Nhieu
+ *
+ */
+public interface Iterator<T> {
+    /**
+     * Returns next element in iteration.
+     * @return  next Object.
+     */
+    public T next();
```
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/bust_a_move/util/Iterator.java]: PMD error -> remove public keyword

File: bust-a-move-framework/src/main/java/bust_a_move/util/Iterator.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    /**
+     * Returns if there is a next element.
+     * @return  true if there is a next element, false otherwise.
+     */
+    public boolean hasNext();
```
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/bust_a_move/util/Iterator.java]: *

File: bust-a-move-framework/src/main/java/bust_a_move/bubble/BubbleStorageIterator.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+ *
+ */
+public class BubbleStorageIterator implements Iterator<Bubble> {
+    private LinkedList<Bubble[]> bubbles;
+    private LinkedListIterator<Bubble[]> listIt;
+    private ArrayIterator<Bubble> rowIt;
+
+    /**
+     * Creates BubbleStorageIterator instance.
+     * @param elements  LinkedList of arrays to iterate over.
```
Comment (calvinnhieu) [inline bust-a-move-framework/src/main/java/bust_a_move/bubble/BubbleStorageIterator.java]: Resolved

File: bust-a-move-framework/src/main/java/bust_a_move/util/Iterator.java (hunk 1 above)
Comment (calvinnhieu) [inline bust-a-move-framework/src/main/java/bust_a_move/util/Iterator.java]: Resolved

File: bust-a-move-framework/src/main/java/bust_a_move/util/Iterator.java (hunk 3 above)
Comment (calvinnhieu) [inline bust-a-move-framework/src/main/java/bust_a_move/util/Iterator.java]: Resolved

**Pull request merged**